### PR TITLE
add email duplicates checks in core users and space users forms

### LINKED ISSUE #278: [Users duplication] forbid usage of same email address for multiple users
forbidding usage of same email address for multiple users will avoid duplication. We have to alert user at account creation

### PR DESCRIPTION
**Pull request by gaetanHerve**
- closes #278 
- merged into form_data_staging - MR #286

### CODE REVIEW

File: Modules/core/Controller/CorespaceaccessController.php [hunk 1]
```diff
@@ -202,12 +202,19 @@ public function useraddAction($id_space){
         $form->setValidationButton(CoreTranslator::Ok($lang), "corespaceaccessuseradd/".$id_space);
 
         if ($form->check()) {
-
             $modelCoreUser = new CoreUser();
+            $canEditUser = true;
+            if ($modelCoreUser->isLogin($this->request->getParameter('login'))) {
+                $canEditUser = false;
+                throw new PfmException(CoreTranslator::LoginAlreadyExists($lang), 403);
```
Comment (osallou) [inline Modules/core/Controller/CorespaceaccessController.php:209]: why throwing an exception here but not in email case?
both should raise an exception or display flash error

File: Modules/core/Controller/CorespaceaccessController.php [hunk 2]
```diff
@@ -202,12 +202,19 @@ public function useraddAction($id_space){
...
+            }
+            if($modelCoreUser->isEmail($form->getParameter("email"))) {
+                // if email alreday exists, warn user
+                $canEditUser = false;
+                $_SESSION["flash"] = CoreTranslator::EmailAlreadyExists($lang);
```
Comment (osallou) [inline Modules/core/Controller/CorespaceaccessController.php:214]: flashClass should be error

File: Modules/core/Controller/CoreusersController.php [hunk 3]
```diff
@@ -125,14 +125,36 @@ public function editAction($id) {
         }
         $script = "";
         if ($form->check()) {
-            if (!$id && $modelUser->isLogin($this->request->getParameter('login'))) {
-                throw new PfmException(CoreTranslator::LoginAlreadyExists($lang), 403);
+            $canEditUser = true;
+            if (!$id) {
+                if ($modelUser->isLogin($this->request->getParameter('login'))) {
+                    $canEditUser = false;
+                    throw new PfmException(CoreTranslator::LoginAlreadyExists($lang), 403);
```
Comment (osallou) [inline Modules/core/Controller/CoreusersController.php:132]: same remark for exception

File: Modules/core/Controller/CoreusersController.php [hunk 4]
```diff
@@ -125,14 +125,36 @@ public function editAction($id) {
...
+                }
+                if($modelUser->isEmail($form->getParameter("email"))) {
+                    // if email alreday exists, warn user
+                    $canEditUser = false;
+                    $_SESSION["flash"] = CoreTranslator::EmailAlreadyExists($lang);
+                }
             } else {
+                $emailUsedByAnother = $modelUser->isEmailWithFilter($form->getParameter("email"), $user["email"]);
```
Comment (osallou) [inline Modules/core/Controller/CoreusersController.php:140]: why a filter ?
if param email == user['email'], that's fine (no change)
if different, should be used nowhere so isEmail should be enough

File: Modules/core/Controller/CorespaceaccessController.php (hunk 1 above)
Comment (gaetanHerve) [reply Modules/core/Controller/CorespaceaccessController.php:209]: It was a PfmException at first for login. I think you put it there. But I prefer flash messages, more user friendly. I push back with a flash message

File: Modules/core/Controller/CoreusersController.php (hunk 4 above)
Comment (gaetanHerve) [reply Modules/core/Controller/CoreusersController.php:140]: ah, yes. It's simpler

File: Modules/core/Controller/CorespaceaccessController.php (hunk 2 above)
Comment (gaetanHerve) [reply Modules/core/Controller/CorespaceaccessController.php:214]: ok, I add that

**Pull request closed without merging**